docs/Guide/preconditions/creating-your-own-preconditions.mdx

```diff
@@ -1,6 +1,228 @@
+[`this.ok()`][preconditionok] to signify the condition has passed, or [`this.error(...)`][preconditionerror] to signify
+the command should be denied.
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript {4-8}
+const { Precondition } = require('@sapphire/framework');
+
+module.exports = class OwnerOnlyPrecondition extends Precondition {
+  public run(message) {
+    return message.author.id === 'YOUR_ID'
+      ? this.ok()
+      : this.error({ message: 'Only the bot owner can use this command!' })
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {4-8}
+import { Precondition } from '@sapphire/framework';
+
+export class OwnerOnlyPrecondition extends Precondition {
+  public run(message) {
+    return message.author.id === 'YOUR_ID'
+      ? this.ok()
+      : this.error({ message: 'Only the bot owner can use this command!' })
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {5-9}
+import type { Message } from '@sapphire/framework';
+import { Precondition } from '@sapphire/framework';
+
+export class OwnerOnlyPrecondition extends Precondition {
+  public run(message: Message) {
+    return message.author.id === 'YOUR_ID'
+      ? this.ok()
+      : this.error({ message: 'Only the bot owner can use this command!' });
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Using Preconditions in Commands
+
+To attach a precondition to a command, you simply have to input its name in an array in the command's
+[`preconditions`][preconditions-option] option.
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript {7}
+const { Command } = require('@sapphire/framework');
+
+module.exports = class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      preconditions: ['OwnerOnly']
+    });
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {7}
+import { Command } from '@sapphire/framework';
+
+export class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      preconditions: ['OwnerOnly']
+    });
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {6}
+import { ApplyOptions } from '@sapphire/decorators';
+import { Command, CommandOptions } from '@sapphire/framework';
+
+@ApplyOptions<CommandOptions>({
+  // ...
+  preconditions: ['OwnerOnly']
+})
+export class PingCommand extends Command {}
+```
+
+</TabItem>
+</Tabs>
+
+Now, if someone who is not the bot owner executes the ping command, nothing will happen!
+
+:::caution
+
+For TypeScript users, there's an extra step to make this work. To increase the security of Sapphire's types, you'll need
+to augment Sapphire's [`Preconditions`][preconditions-interface] interface. Please see an official example
+[here][preconditions-augment].
```

Comment:
```suggestion
[here][preconditions-augment].

<!-- TODO: Once we have a dedicated TypeScript page this should link to that instead of a code example -->
```

So we have a reminder to add it later